### PR TITLE
Authentication: Members Only Project

### PR DESCRIPTION
**Pull request by nriqu322**
This is an authentication project where only logged users can see the author of the gossip.

- We use the rails generator to create the skeleton of the app
- Create a users and posts model
- Devise gem used to authenticate users and create sessions
- Use Bulma for little styling
- We use Rubocop and stickler to lint

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 Rails.application.routes.draw do
+  devise_for :users, controllers: { registrations: 'registrations' }
   resources :posts
```
Comment (maelfosso) [inline config/routes.rb:3]: It was asked to set it for 3 routes only

Screenshot_2020-04-22_21-23-11

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
   def index
     @posts = Post.all
+    @post = Post.new
   end
 
   # GET /posts/1
   # GET /posts/1.json
-  def show
-  end
+  def show; end
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb:14]: Not needed

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,35 +1,35 @@
...
   def new
-    @post = Post.new
+    @post = current_user.posts.build
+    # @post = Post.new
   end
 
   # GET /posts/1/edit
-  def edit
-  end
+  def edit; end
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb:23]: Not needed

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -1,35 +1,35 @@
...
   # GET /posts/1.json
-  def show
-  end
+  def show; end
 
   # GET /posts/new
   def new
-    @post = Post.new
+    @post = current_user.posts.build
+    # @post = Post.new
```
Comment (maelfosso) [inline app/controllers/posts_controller.rb:19]: Remove comment

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (nriqu322) [reply app/controllers/posts_controller.rb]: removed

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (nriqu322) [reply app/controllers/posts_controller.rb]: removed

File: app/controllers/posts_controller.rb (hunk 4 above)
Comment (nriqu322) [reply app/controllers/posts_controller.rb]: removed

File: config/routes.rb (hunk 1 above)
Comment (nriqu322) [reply config/routes.rb]: added only %i[new create index]

**Pull request merged**